#22 지역 조회 캐쉬 적용

0hun opened the pull request:
지역 조회시 캐시 적용하는 어노테이션 추가

0hun committed:
#22 지역 조회 캐쉬 적용

지역 조회시 캐시 적용하는 어노테이션 추가

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -28,6 +29,11 @@
+   * @Cacheable : 동일 값이 Cache에 있는 경우 Cache에서 데이터를 return합니다.
+   * 만약 동일 key 값이 없을 경우 메소드를 실행하고 반환된 결과 값을 Cache에 저장합니다.
+   */
+  @Cacheable(value = "regions", key = "#name")
```
만료시간 같은 정책들도 정해줄 수 있을 것 같네요

f-lab-dev commented on src/main/java/com/younghun/hibusgo/HiBusGoApplication.java:
```diff
@@ -2,7 +2,12 @@
+import org.springframework.cache.annotation.EnableCaching;
 
+/**
+ * @EnableCaching : Application에게 Cache를 사용하다는 의미로 사용
```
이전 PR과 중복인것 같은데 이거는 브랜치를 develop에서 만드는게 아니라 이전 PR 브랜치에서 만들었어야했을 것 같네요~

0hun committed:
#22 리뷰 반영

redis cache 만료시간 추가

f-lab-dev commented on src/main/resources/application.properties:
```diff
@@ -23,4 +23,5 @@ spring.redis.database=0
-spring.redis.timeout=60000
\ No newline at end of file
+spring.redis.timeout=60000
+spring.cache.redis.time-to-live=60000
```
이 설정은 글로벌설정으로 보입니다. 그런데 각 기능별로 최적의 타임아웃이 다를 것 같은데요.
어떻게 따로 설정해줄 수 있을까요?
메모리를 오버하는 것을 방지하기 위해 최대 크기 제한도 두면 좋을 것 같고요~

0hun committed:
Merge branch 'develop' into issue/22

0hun committed:
#22 리뷰 반영

RedisConfig에 redis 설정 추가
CacheManager로 각 기능별 설정 추가
resultMap 잘못 설정되있어서 변경
accont가 변경되어 오류가 발생하여 수정

0hun committed:
Merge branch 'issue/22' of https://github.com/f-lab-edu/hi-bus-go into issue/22

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -0,0 +1,82 @@
+   *
+   * Lettuce - Netty 기반 redis client library
+   * - 비동기로 요청하기 때문에 Jedis에 비해 높은 성능을 가지고 있다.
+   * - TPS, 자원사용량 모두 Jedis에 비해 우수한 성능을 보인다는 테스트 사례가 있다.
```
netty, blocking io, nonblocking io 에 대해 알아보시면 좋을 것 같습니다

0hun replied:
넵 알겠습니다.

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -0,0 +1,82 @@
+        .serializeValuesWith(RedisSerializationContext
+            .SerializationPair
+            .fromSerializer(new GenericJackson2JsonRedisSerializer()))
+        .entryTtl(Duration.ofSeconds(1800));
```
캐시가 1800초동안 지속되면 트래픽이 많을 때 메모리가 꽉 차는 현상이 나타나지 않을까요?

0hun committed:
#22 리뷰 반영

캐시 지속 시간이 30분일 경우 트래픽이 많으면 메모리가 꽉 찰수도 있어서
5분으로 수정
